packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js

```diff
@@ -0,0 +1,810 @@
+    network,
+    rpcURL,
+    iterBlocksFetchTimeoutMS,
+  });
+  beforeEach(() => {
+    provider = new NEOONEDataProvider({
+      network,
+      rpcURL,
+      iterBlocksFetchTimeoutMS,
+    });
+  });
+
+  test('getUnclaimed', async () => {
+    const expected = { unclaimed: [], amount: new BigNumber(0) };
+    // $FlowFixMe
+    provider._client.getAccount = jest.fn(() =>
+      Promise.resolve({
+        unclaimed: [],
+      }),
+    );
+    // $FlowFixMe
+    provider._client.getClaimAmount = jest.fn(() =>
+      Promise.resolve([new BigNumber(0)]),
+    );
+
+    const result = await provider.getUnclaimed(keys[0].address);
+    expect(result).toEqual(expected);
+  });
+
+  test('getUnspentOutputs', async () => {
+    const expected = [
+      {
+        asset: '',
+        value: new BigNumber('0'),
+        address: '',
+        txid: '',
+        vout: 0,
+      },
+    ];
+    // $FlowFixMe
+    provider._client.getAccount = jest.fn(() =>
+      Promise.resolve({
+        unspent: [
+          {
+            txid: '',
+            vout: 0,
+          },
+          null,
+        ],
+      }),
+    );
+    // $FlowFixMe
+    provider._client.getUnspentOutput = jest
+      .fn()
+      .mockReturnValueOnce(
+        Promise.resolve({
+          asset: '',
+          value: '0',
+          address: '',
+        }),
+      )
+      .mockReturnValueOnce(null);
+
+    const result = await provider.getUnspentOutputs(keys[0].address);
+    expect(result).toEqual(expected);
+    expect(provider._client.getUnspentOutput).toHaveBeenCalledTimes(2);
+  });
+
+  describe('_ConvertTransactionBase types', () => {
+    const testCases = [
+      {
+        expected: genExpectedTransaction('MinerTransaction', { nonce: 10 }),
+        transactionJSON: genTransactionJSON('MinerTransaction', { nonce: 10 }),
+      },
+      {
+        expected: genExpectedTransaction('ClaimTransaction', { claims: [] }),
+        transactionJSON: genTransactionJSON('ClaimTransaction', { claims: [] }),
+      },
+      {
+        expected: genExpectedTransaction('ContractTransaction', {}),
+        transactionJSON: genTransactionJSON('ContractTransaction', {}),
+      },
+      {
+        expected: genExpectedTransaction('EnrollmentTransaction', {
+          publicKey: '0',
+        }),
+        transactionJSON: genTransactionJSON('EnrollmentTransaction', {
+          pubkey: '0',
+        }),
+      },
+      {
+        expected: genExpectedTransaction('IssueTransaction', {}),
+        transactionJSON: genTransactionJSON('IssueTransaction', {}),
+      },
+      {
+        expected: genExpectedTransaction('PublishTransaction', {
+          contract: genExpectedContract({}),
+        }),
+        transactionJSON: genTransactionJSON('PublishTransaction', {
+          contract: genContractJSON({}),
+        }),
+      },
+      {
+        expected: genExpectedTransaction('RegisterTransaction', {
+          asset: (({ type, name, amount, precision, owner, admin }) => ({
```

Comment:
Extract this logic out into a common function, like `createRegisterExpectedAsset`.

Reply:
Also, as a small nit, it took me a second to figure out what was going on here - creating an anonymous function just to extract some of the properties is not a very common pattern.